src/inspect_ai/_cli/score.py

```diff
@@ -64,6 +65,12 @@
     type=click.Path(dir_okay=False, writable=True),
     help="Output file to write the scored log to.",
 )
+@click.option(
+    "--stream",
+    type=bool,
+    is_flag=True,
+    help="Stream the samples through the scoring process instead of reading the entire log into memory. Useful for large logs.",
+)
```

Comment:
It would probably be nice to make `--stream` accept and integer argument, which is the maximum number of samples that would be read into memory at any one time. That said, I consider the implementation in `score` a reference implementation. I imagine the most useful loader would be a bit more dynamic and available system memory.